frontend/src/reducer.test.ts

```diff
@@ -0,0 +1,173 @@
+import { nanoid } from 'nanoid';
+import MessageChain, { MessageType } from './classes/MessageChain';
+import Player from './classes/Player';
+import TownsServiceClient from './classes/TownsServiceClient';
+import { CoveyAppState } from './CoveyTypes';
+import { appStateReducer } from './reducer';
+import { createMessageForTesting } from './TestUtils';
+
+jest.mock('./classes/TownsServiceClient');
+
+const createSampleAppState = (): CoveyAppState => ({
+  nearbyPlayers: { nearbyPlayers: [] },
+  players: [],
+  myPlayerID: '123',
+  currentTownFriendlyName: '',
+  currentTownID: '',
+  currentTownIsPubliclyListed: false,
+  sessionToken: '',
+  userName: '',
+  socket: null,
+  currentLocation: {
+    x: 0,
+    y: 0,
+    rotation: 'front',
+    moving: false,
+  },
+  emitMovement: () => {},
+  apiClient: new TownsServiceClient(),
+  townMessageChain: new MessageChain(),
+  proximityMessageChain: new MessageChain(),
+  directMessageChains: {},
+});
+
+describe('reducer', () => {
+  const player1Id = '123';
+  const player2Id = '456';
+  const directMessageId = '123:456';
+
+  describe('messageReceived', () => {
+    it('adds new messages to existing town message chain', () => {
+      const messageToTest = createMessageForTesting(MessageType.TownMessage, nanoid());
```

Comment:
Nit: keeping with the patterns of the other test variables, this should be `message1ToTest`